services/frontend-service/src/ui/Pages/Home/Home.test.tsx

```diff
@@ -32,20 +32,278 @@ describe('App', () => {
     const getWrapper = () => render(getNode());
     it('Renders full app', () => {
         // when
+        const appBuilder = (suffix: string) => ({
+            name: 'test' + suffix,
+            releases: [],
+            sourceRepoUrl: 'http://test' + suffix + '.com',
+            team: 'example' + suffix,
+        });
         UpdateOverview.set({
             environments: {},
             applications: {
-                app1: {} as any,
-                app3: {} as any,
-                app2: {} as any,
+                app1: appBuilder('1'),
+                app3: appBuilder('3'),
+                app2: appBuilder('2'),
             },
         } as any);
         getWrapper();
 
         // then apps are sorted and Service Lane is called
-        expect(mock_ServiceLane.ServiceLane.getCallArgument(0, 0)).toStrictEqual({ application: 'app1' });
-        expect(mock_ServiceLane.ServiceLane.getCallArgument(1, 0)).toStrictEqual({ application: 'app2' });
-        expect(mock_ServiceLane.ServiceLane.getCallArgument(2, 0)).toStrictEqual({ application: 'app3' });
+        expect(mock_ServiceLane.ServiceLane.getCallArgument(0, 0)).toStrictEqual({ application: appBuilder('1') });
+        expect(mock_ServiceLane.ServiceLane.getCallArgument(1, 0)).toStrictEqual({ application: appBuilder('2') });
+        expect(mock_ServiceLane.ServiceLane.getCallArgument(2, 0)).toStrictEqual({ application: appBuilder('3') });
+    });
+});
+
+describe('Application Filter', () => {
+    interface dataT {
+        name: string;
+        query: string;
+        applications: string[];
+        expect: (nrLocks: number) => void;
+    }
+
+    const data: dataT[] = [
+        {
+            name: 'filter applications - 1 result',
+            applications: ['dummy', 'test', 'test2', 'foo'],
+            query: 'dummy',
+            expect: (nrLocks) => expect(nrLocks).toStrictEqual(1),
```

Comment:
Can be simplified:
```suggestion
            expectedLocks: 1,
```
And then do the `expect` call in the actual test (line 91)

Reply:
see KB entry: https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/WORKSPACE_DOCUMENT/-N-NHZSPe5bbFj8eU1Bz/
(not exactly the same, but same principle applies)

Reply:
Thank you for the tip! Now the tests are much more readable and even easier to write! :)
Fixed!